docs: adding details on how to declare a UIB app and other improvements (#2773)
bonitasoft/bonita-doc: modules/applications/pages/builder-declare-interface-in-bonita.adoc

There are fewer properties to set for an application link than for a legacy application, but the `displayName` is still mandatory.

> The `token` property's value must be identical to the value of the `slug` key found in the xref:package-and-deploy-your-application.adoc#_export_your_application[JSON export of your application]. It is usually identical to your application's name.

Reviewer: > It is usually identical to your application's name.

Not really, the name is converted to kebab-case. So in your example "demo_application" was converted to "demo-application". Another example is "Demo Application" will be converted to "demo-application" as well.
To prevent confusion, I would not say it is identical to the application name.

Author: ```suggestion
The `token` property's value must be identical to the value of the `slug` key found in the xref:package-and-deploy-your-application.adoc#_export_your_application[JSON export of your application]. For example, if your application name is _Awesome application_, its `slug` should be _awesome-application_.
```

Looks better?

> The `token` property's value must be identical to the value of the `slug` key found in the xref:package-and-deploy-your-application.adoc#_export_your_application[JSON export of your application]. For example, if your application name is _Awesome application_, its `slug` should be _awesome-application_.

Reviewer: Italic syntax is not correctly formatted
```suggestion
The `token` property's value must be identical to the value of the `slug` key found in the xref:package-and-deploy-your-application.adoc#_export_your_application[JSON export of your application]. For example, if your application name is __Awesome application__, its `slug` should be __awesome-application__.
```
